client/tests/unit/HMFForm.spec.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-await-in-loop, no-restricted-syntax, no-continue */
+
+import { mount, createLocalVue } from '@vue/test-utils';
+import HMFForm from '@/components/HMFForm';
+import BACKEND_CONSTANTS from '@/constants/backend_constants';
+
+describe('Mounted HMFForm', () => {
+  const localVue = createLocalVue();
+  const wrapper = mount(HMFForm, localVue);
+
+  const options = Object.keys(BACKEND_CONSTANTS.FittingFunction_params);
+
+  test('has correct default model', () => {
+    expect(wrapper.vm.model.hmf_model).toBe('Tinker08');
+  });
+
+  test('changes model parameters when model is changed', async () => {
+    for (const option of options) {
+      if (wrapper.vm.model.hmf_model === option) continue;
+      const oldParams = wrapper.vm.model.hmf_params;
+      wrapper.vm.$data.model.hmf_model = option;
+      await localVue.nextTick();
+      await localVue.nextTick();
+      const newParams = wrapper.vm.model.hmf_params;
+      expect(JSON.stringify(oldParams)).not.toBe(JSON.stringify(newParams));
+    }
+  });
+
+  test('renders correct fields for each model selection', async () => {
+    for (const option of options) {
+      wrapper.vm.$data.model.hmf_model = option;
+      await localVue.nextTick();
+      await localVue.nextTick();
+      const params = Object.keys(BACKEND_CONSTANTS.FittingFunction_params[option]);
+      for (const param of params) {
+        expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp(`.*${param}.*`)));
+      }
+    }
+  });
+
+  // leaving this in as documentation for another way to potentially do this: dynamically generates
+  // narrower test cases to give you a better idea of the point of failure (thoughts?)
+  //
+  // test.each(options)('renders correct fields for %s model selection', async (option) => {
+  //     wrapper.vm.$data.model.hmf_model = option;
+  //     await localVue.nextTick();
+  //     await localVue.nextTick();
+  //     const params = Object.keys(BACKEND_CONSTANTS.FittingFunction_params[option]);
+  //     for (const param of params) {
+  //       expect(wrapper.html()).toEqual(expect.stringMatching(new RegExp(`.*${param}.*`)));
+  //     }
+  // })
```

Comment:
Ya, this seems like a great idea. Keeping the code cleaner would be good, as long as we don't end up testing the tests, this seems like a great reduction in the overall amount of code.